### ISSUE TITLE
Bug: VPN cannot add route  for destination ::/0: ... : no such device

### ISSUE THREAD

**Issue by Eoin-ONeill-Yokai** [qdm12/gluetun#2597]
### Is this urgent?

None

### Host OS

Fedora CoreOS

### CPU arch

x86_64

### VPN service provider

NordVPN

### What are you using to run the container

Podman

### What is the version of Gluetun

Running version latest built on 2024-11-18T09:49:16.711Z (commit 68ddbfc)

### What's the problem 🤔

Seems to me that it can't write to the tunnel device for IPv6, which causes an infinite fail loop on first boot. Having said that, I'm not positive it isn't a configuration issue yet. Any help or advice from someone who got this working in a podman environment would be helpful!

Other possible issues:
- `firewalld` might be getting in the way. Is there any advice for configuring firewalld for use of gluetun via podman container?

### Share your logs (at least 10 lines)

```plain text
gluetun  | 2024-11-23T23:37:54-08:00 ERROR [vpn] cannot add route for interface: adding route for destination ::/0: adding route for link tun0, destination ::/0 and table 51820: no such device
gluetun  | 2024-11-23T23:37:54-08:00 INFO [vpn] retrying in 15s
gluetun  | 2024-11-23T23:37:54-08:00 DEBUG [wireguard] deleting link...
gluetun  | 2024-11-23T23:38:00-08:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (healthcheck error: dialing: dial tcp4: lookup cloudflare.com on 1.1.1.1:53: write udp 10.89.1.12:55792->1.1.1.1:53: write: operation not permitted)
gluetun  | 2024-11-23T23:38:00-08:00 INFO [healthcheck] 👉 See https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md
gluetun  | 2024-11-23T23:38:00-08:00 INFO [healthcheck] DO NOT OPEN AN ISSUE UNLESS YOU READ AND TRIED EACH POSSIBLE SOLUTION
````

^ Basically this on loop, but I imagine that first error is probably the source of the problem.
```


### Share your configuration

```yml
services:
  gluetun:
    image: ghcr.io/qdm12/gluetun
    container_name: gluetun
    # line above must be uncommented to allow external containers to connect.
    # See https://github.com/qdm12/gluetun-wiki/blob/main/setup/connect-a-container-to-gluetun.md#external-container-to-gluetun
    cap_add:
      - NET_ADMIN
      - NET_RAW
      - MKNOD # < Unsure about the necessity of this
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 8888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
    volumes:
      - ./gluetun:/gluetun:z
    environment:
      # See https://github.com/qdm12/gluetun-wiki/tree/main/setup#setup
      - VPN_SERVICE_PROVIDER=nordvpn
      - VPN_TYPE=wireguard
      # OpenVPN:
      # - OPENVPN_USER=
      # - OPENVPN_PASSWORD=
      # Wireguard:
      - WIREGUARD_PRIVATE_KEY=NopeNope:)
      - WIREGUARD_MTU=1000
      - SERVER_COUNTRIES=United States,Ireland
      # Timezone for accurate log times
      - TZ=America/Los_Angeles
      - LOG_LEVEL=debug
      # Server list updater
      # See https://github.com/qdm12/gluetun-wiki/blob/main/setup/servers.md#update-the-vpn-servers-list
      - UPDATER_PERIOD=24h
    networks:
      - isolated
    restart: unless-stopped
    healthcheck:
      test: "ping -c 1 google.com || exit 1"
      timeout: 2s
      retries: 3
    security_opt:
      - label=disable # Also unsure about the necessity of this
```

**Comment by qdm12** [comment]
`no such device` may mean `tun0` doesn't exit - weird.
Please try image tag `:pr-2629` to see if it helps, it now specifies the ip family for the route to add, depending on the destination. Maybe the netlink library I use or kernel is not smart enough to figure this out from the destination ip prefix specified.

Also please report the logs you get with `LOG_LEVEL=debug` thanks!

**Comment by ToxicMushroom** [comment]
fwiw nordvpn doesn't support ipv6, not sure if it's related at all here

**Comment by agittins** [comment]
I'm connecting to airvpn, running on a vm with 5.10.0-34-cloud-amd64 and the wireguard connection stopped working for me back in Sept 2024, so I switched to openvpn. At the time I tried the various `--device`, priv options without success.

I came at it again today after doing a mobo replacement, and was still having the same issue, with `ERROR [vpn] cannot add route for interface: adding route for destination ::/0: adding route for link tun0, destination ::/0 and table 51820: no such device`.

Using the `:pr-2629` image solved the issue for me, thanks!

Unsure if it's relevant, but I'm using the envar `- WIREGUARD_ADDRESSES=10.xxx.xx.xx/32` [redacted] option, ie not including an ipv6 address (a change I made in 2023-12-16 after an update to get things working).

@Eoin-ONeill-Yokai have you tried changing to `image: qmcgaw/gluetun:pr-2629`?

**Comment by Eoin-ONeill-Yokai** [comment]
Sorry for the late response, but I haven't tried this yet. I'll give it a chance the next time I'm doing some server tinkering.

**Comment by qdm12** [comment]
Finally merged, sorry for the massive delay, life got in the way! If you face another similar issue, please create another issue!
